alert-sim/dataclass/FacilityControl.py

```diff
@@ -0,0 +1,22 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Optional
+from constants import Faction
+from .MapControl import MapControl
+
+@dataclass
+class FacilityControl:
+    instance: str
+    facilityId: int
+    timestamp: datetime
+    oldFaction: Faction
+    newFaction: Faction
+    duration_held: int = 0
+    isInitial: bool = False
+    isDefence: bool = False
+    outfitCaptured: Optional[int] = None
+    mapControl: Optional[MapControl] = None
+
+    def __post_init__(self):
+        self.timestamp = self.timestamp.isoformat()
```

Comment:
I would not recommend converting this to a string if the type on the class is datetime

Reply:
@ryanjsims unfortunately it breaks JSON encoding so we have no real choice